python/pdstools/adm/Reports.py

```diff
@@ -0,0 +1,417 @@
+        *,
+        output_type: str = "html",
+        keep_temp_files: bool = False,
+        verbose: bool = False,
+    ) -> Path:
+        """
+        Generates Health Check report based on the provided parameters.
+
+        Parameters
+        ----------
+        name : str, optional
+            The name of the report.
+        working_dir : Union[str, Path, None], optional
+            The working directory for the output. If None, uses current working directory.
+        output_type : str, default='html'
+            The type of the output file (e.g., "html", "pdf").
+        keep_temp_files : bool, optional
+            If True, the temporary directory with temp files will not be deleted after report generation.
+        verbose: bool, optional
+            If True, prints detailed logs during execution.
+
+        Returns
+        -------
+        Path
+            The path to the generated report file.
+
+        Raises
+        ------
+        ValueError
+            If there's an error in report generation or invalid parameters.
+        FileNotFoundError
+            If required files are not found.
+        subprocess.SubprocessError
+            If there's an error in running external commands.
+        """
+        working_dir, temp_dir = cdh_utils.create_working_and_temp_dir(name, working_dir)
+        try:
+            qmd_file = "HealthCheck.qmd"
+            output_filename = self._get_output_filename(
+                name, "HealthCheck", None, output_type
+            )
+
+            self._copy_quarto_file(qmd_file, temp_dir)
+            model_file_path, predictor_file_path = self.datamart.save_data(temp_dir)
+            params = {
+                "report_type": "HealthCheck",
+                "model_file_path": str(model_file_path),
+                "predictor_file_path": str(predictor_file_path),
+            }
+
+            self._write_params_file(temp_dir, params)
+            self._run_quarto_command(
+                temp_dir,
+                qmd_file,
+                output_type,
+                output_filename,
+            )
+
+            output_path = temp_dir / output_filename
+            if not output_path.exists():
+                raise ValueError(f"Failed to generate report: {output_filename}")
+
+            final_path = working_dir / output_filename
+            shutil.copy(output_path, final_path)
+            return final_path
+
+        finally:
+            if not keep_temp_files and temp_dir.exists() and temp_dir.is_dir():
+                shutil.rmtree(temp_dir, ignore_errors=True)
+
+    def _get_output_filename(self, name, report_type, model_id, output_type):
+        """Generate the output filename based on the report parameters."""
+        name = name.replace(" ", "_") if name else None
+        if report_type == "ModelReport":
+            if not model_id:
+                raise ValueError("model_id is required for a model report.")
```

Comment:
Model ID is a required argument to this function? I don't understand how it could be missing. Sidenote: let's annotate these arguments

Reply:
It was indeed unnecessary, removed this part.